### ISSUE TITLE
Applying Jordan-Wigner to FermionOperator vs. InteractionOperator

### ISSUE THREAD

**Issue by max-radin** [quantumlib/OpenFermion#479]
In some cases the result of a Jordan-Wigner transform applied to an InteractionOperator is different if it is converted to a FermionOperator first, as shown in the code below. It seems like `jordan_wigner_interaction_op` only gives the correct result for Hermitian operators. 

```python
from openfermion.ops import InteractionOperator
from openfermion.transforms import get_fermion_operator, jordan_wigner
import numpy as np

N = 4
kappa = np.zeros((N, N))
kappa[0,3] = 2.0
op = InteractionOperator(0., -kappa, np.zeros((N, N, N, N)))

qubit_op1 = jordan_wigner(op)
print(qubit_op1)
# -0.5 [X0 Z1 Z2 X3] +
# -0.5 [Y0 Z1 Z2 Y3]

qubit_op2 = jordan_wigner(get_fermion_operator(op))
print(qubit_op2)
# (-0.5+0j) [X0 Z1 Z2 X3] +
# -0.5j [X0 Z1 Z2 Y3] +
# 0.5j [Y0 Z1 Z2 X3] +
# (-0.5+0j) [Y0 Z1 Z2 Y3]
```

**Comment by kevinsung** [comment]
Indeed, `jordan_wigner_interaction_op` is only guaranteed to work for Hermitian operators and is optimized based on that assumption. Indeed, the InteractionOperator class is intended to represent only Hermitian operators, even though this is neither documented nor enforced. @babbush or @jarrodmcc can correct me if I'm wrong. I think we need to document this better; enforcement would be tricky, though there are certainly things we can do to help (e.g. #444 ).

**Comment by matti-sarjala** [comment]
A small documentation clarification could make this boundary much easier for users to understand.

The useful distinction seems to be:

- `FermionOperator` represents a general algebraic fermionic operator.
- `InteractionOperator` represents a structured electronic-structure Hamiltonian, with Hermitian assumptions built into the representation and optimized transforms.

So the two routes are not guaranteed to be interchangeable for arbitrary non-Hermitian input:

1. `jordan_wigner(FermionOperator(...))`
2. `jordan_wigner(InteractionOperator(...))`

A minimal docs note could say something like:

`InteractionOperator` is intended for Hermitian molecular/electronic-structure Hamiltonians. For general non-Hermitian fermionic operators, use `FermionOperator` before applying transforms such as Jordan-Wigner.

It may also help to add one small example or warning showing that an off-diagonal one-body tensor without the Hermitian conjugate term is not a valid `InteractionOperator` use case. That would prevent users from interpreting the optimized transform as a general replacement for the full symbolic transform.

**Comment by mhucka** [comment]
Fixed by https://github.com/quantumlib/OpenFermion/pull/1306 .
